xdsl/irdl.py

```diff
@@ -16,6 +16,8 @@
 from xdsl.ir import (Attribute, Block, Data, OpResult, Operation,
                      ParametrizedAttribute, Region, SSAValue)
 
+# pyright: reportMissingParameterType=false, reportUnknownParameterType=false
```

Comment:
Why do we need this? Becomes some of the reflection magic in this file is not introspectable?

Reply:
I would guess this is because Python does not know how to type lambdas, which was creating a few warnings since we use lambdas a lot here.